CMakeLists.txt

```diff
@@ -960,10 +960,25 @@ IF (WITH_H2OLOG)
 ENDIF (WITH_H2OLOG)
 
 # gitrev
-IF ("${VERSION_PRERELEASE}" STREQUAL "-DEV" AND EXISTS "${CMAKE_SOURCE_DIR}/.git")
+SET(GITREV_IS_AVAILABLE FALSE)
+IF (EXISTS "${CMAKE_SOURCE_DIR}/.git")
+    # asks generate_gitrev.pl whether or not gitrev is available.
+    EXECUTE_PROCESS(
+        COMMAND perl ${CMAKE_CURRENT_SOURCE_DIR}/misc/generate_gitrev.pl
+        WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+        RESULT_VARIABLE GITREV_RESULT_CODE
+        OUTPUT_QUIET
+        ERROR_QUIET
+    )
+    IF (GITREV_RESULT_CODE EQUAL 0)
+        SET(GITREV_IS_AVAILABLE TRUE)
+    ENDIF ()
+ENDIF ()
```

Comment:
I think we can move this check (and potentially the initial generation of gitrev.h) to the IF statement below so that it would only be run when it is a dev release?

Directory access constraints can be rigid for package maintainers that build the release versions.